application/src/main/java/org/thingsboard/server/service/subscription/DefaultTbLocalSubscriptionService.java

```diff
@@ -176,6 +177,10 @@ public void onCoreStartupMsg(TransportProtos.CoreStartupMsg coreStartupMsg) {
         });
     }
 
+    Lock getSubsLock(TenantId tenantId) {
```

Comment:
I suggest adding a class *EntityLocksStorage* that will use the *ConcurrentReferenceHashMap* and encapsulate a single operation with the locks—get(EntityId id). You will be able to cover it with tests (if needed). Then, the code will be a bit simpler, and we may reuse it in the future. Also, you will not need the get methods without the *private* modifier.